python/cudf/cudf/tests/test_datetime.py

```diff
@@ -253,6 +283,25 @@ def test_typecast_to_different_datetime_resolutions(data, dtype):
     np.testing.assert_equal(np_data, np.array(gdf_series))
 
 
+@pytest.mark.parametrize(
+    "data", [timestamp_ms_data(), timestamp_us_data(), timestamp_ns_data()]
+)
+@pytest.mark.parametrize(
+    "dtype",
+    ["datetime64[s]", "datetime64[ms]", "datetime64[us]", "datetime64[ns]"],
+)
+def test_string_timstamp_typecast_to_different_datetime_resolutions(
+    data, dtype
+):
+    pd_sr = data
+    gdf_sr = cudf.Series.from_pandas(pd_sr)
+
+    expect = np.array(pd_sr.astype(dtype))
```

Comment:
Pandas only supports `datetime64[ns]` so this doesn't effectively test the code. You should use numpy or pyarrow for the CPU codepath here.